### PR TITLE
SLM-252 supported prisons api

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

**Review by hyderkhanmoj — COMMENTED** [review]
Could use JPAs 'OrderBy' on the repository instead of ordering in the service layer - same result at the end

**Review by hyderkhanmoj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisons/SupportedPrisonsRepository.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@Entity
+@Table(name = "supported_prisons")
+data class SupportedPrison(
+  @Id
+  val code: String = "",
+  val active: Boolean,
+  val updatedBy: String,
+  val updated: Instant,
+) {
+  override fun equals(other: Any?): Boolean {
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisons/SupportedPrisonsRepository.kt:26]: Just wondering, why override  hasCode and equals for these data classes?

**Review by yasinmustafa — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisons/SupportedPrisonsRepository.kt (hunk 1 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisons/SupportedPrisonsRepository.kt:26]: It's to do with Hibernate and storing entities in collections - basically if you update an entity then add it to a collection it should replace the existing entity in that collection, not add another element. There's a good article about it [here](https://vladmihalcea.com/the-best-way-to-implement-equals-hashcode-and-tostring-with-jpa-and-hibernate/).

Intellij issues a warning for JPA entities if you don't override equals and hashcode... I use a plugin called JPA Buddy to generate the equals and hascode for me.

**Pull request merged**